routetable/usage_manager.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routetable
+
+import (
+	"errors"
+
+	"github.com/golang-collections/collections/stack"
+	v3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/set"
+)
+
+type RouteTableManager struct {
```

Comment:
Instead of `Manager`, how about `Allocator`; Manager is always a bit woolly (wish I had a more specific name for the endpoint manager et al)

Reply:
Yes, have just pushed a change to make this more generic, named IndexAllocator.